strawberry/ext/mypy_plugin.py

```diff
@@ -388,13 +390,53 @@ def strawberry_pydantic_class_callback(ctx: ClassDefContext) -> None:
         ]
         add_method(ctx, "__init__", init_args, NoneType())
 
-        model_type = _get_type_for_expr(model_expression, ctx.api)
+        model_type: mypy.types.Instance = _get_type_for_expr(model_expression, ctx.api)
+
+        # these are the fields that the user added to the strawberry type
+        new_strawberry_fields: Set[str] = set()
+
+        # TODO: think about inheritance for strawberryt?
+        for stmt in ctx.cls.defs.body:
```

Comment:
theres a number of TODOs here, but i'm submitting this MR since a number of people are requesting it
https://discord.com/channels/689806334337482765/773519351423827978/989937871450411080